core/trino-spi/src/main/java/io/trino/spi/connector/EntityKindAndName.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public record EntityKindAndName(String entityKind, @JsonProperty List<String> name)
+{
+    public EntityKindAndName
+    {
+        requireNonNull(entityKind, "entityKind is null");
+        name = List.copyOf(requireNonNull(name, "name is null"));
```

Comment:
requireNonNull not needed for this one. List.copyOf will fail if `name` is null.

Reply:
Removed.